From 07c842e83fd7e8d2e392ee7421f4abd037554c64 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 27 May 2021 14:35:54 +0300
Subject: [PATCH] Revert "Revert "[SW-41316] preboot: make flexible fw load setting platform specific""

This reverts commit 6febf9854c41c710177ac1b0f29d0ed585ab2738.

In dynamic firmware load flow, dram scrambling is still performed
after linux fit load by LKD causing issue in firmware load. Till
this issue is fixed, disable dynamic fw load for greco.

Change-Id: Iadca8a444242f3d08e18571fc7b8ab75c5df51cd
---

diff --git a/gaudi/common.c b/gaudi/common.c
index 745079f..f1e3e7c 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -444,6 +444,7 @@
 void lkd_dev_specific_setup(void)
 {
 	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_HARD_RST_EN, true);
+	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_LD_COM_EN, true);
 }
 
 /* Hard coded each of the gaudi on-chip TS device properties */
diff --git a/goya/common.c b/goya/common.c
index 6488f2d..9e7d889 100644
--- a/goya/common.c
+++ b/goya/common.c
@@ -25,6 +25,7 @@
 #include <preboot/fit.h>
 #include <preboot/thermal.h>
 #include <string.h>
+#include <boot_if.h>
 
 #include "common.h"
 
@@ -218,6 +219,11 @@
 	return mmCPU_TIMESTAMP_CNTCR;
 }
 
+void lkd_dev_specific_setup(void)
+{
+	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_LD_COM_EN, true);
+}
+
 #define GOYA_ASIC_TS_MAX_COUNT 3
 
 int hl_ts_get_count(void)
diff --git a/preboot/module_lkd.c b/preboot/module_lkd.c
index a6aba2c..053716d 100644
--- a/preboot/module_lkd.c
+++ b/preboot/module_lkd.c
@@ -32,7 +32,6 @@
 
 void module_lkd_enable(void)
 {
-	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_LD_COM_EN, true);
 	cpu_boot_sts_set(CPU_BOOT_STATUS_WAITING_FOR_BOOT_FIT);
 }
 
